flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/delegation/PlannerBase.scala

```diff
@@ -455,16 +455,21 @@ abstract class PlannerBase(
    * the configuration before planner do optimization with [[ModifyOperation]] or other works.
    */
   protected def validateAndOverrideConfiguration(): Unit = {
-    if (!config.getConfiguration.get(TableConfigOptions.TABLE_PLANNER).equals(PlannerType.BLINK)) {
+    val configuration = config.getConfiguration
+    if (!configuration.get(TableConfigOptions.TABLE_PLANNER).equals(PlannerType.BLINK)) {
       throw new IllegalArgumentException(
         "Mismatch between configured planner and actual planner. " +
           "Currently, the 'table.planner' can only be set when instantiating the " +
           "table environment. Subsequent changes are not supported. " +
           "Please instantiate a new TableEnvironment if necessary.");
     }
 
+    // Add a query start time to TableConfig, this config is only used internal,
+    // this config will be used by temporal functions like CURRENT_TIMESTAMP.
+    configuration.setLong("__table.query-start.epoch-time__", System.currentTimeMillis())
```

Comment:
Is there a possibility of removing the flag again after submission? Otherwise I think this approach is good enough to have query-start semantics.

Reply:
Thanks for your feedback, I'll remove this flag